### DISCUSSION TITLE
100% maximum width and height to fill screen?

### DISCUSSION THREAD

**Discussion by chip33** [Hofknecht/SystemTrayMenu#432] [Ideas]
I appreciate how much time and work coding this application must be.
The changes lately have been amazing and each new release is a huge step forward.
I have an idea but I do not wish to ask for coding work unless you agree it is a good improvement and feel it is worth while.

Could the Maximum menu height at 100% be a variable (just for the root menu GUI) that is calculated from the bottom left corner of STM to the top of the screen in pixels?
Then (allowing for title bar or function bar) the STM window height would maximise to fit perfectly onto any display wherever it is positioned.
Sub folders already calculate their size to completely fit the screen height when Maximum menu height is larger than screen height (>125%), could they be fitted to the screen height when Maximum menu height is 100%?

Maybe something similar could also be done with maximum menu width at 100% being the distance in pixels from bottom left corner of STM to screen edge on right just for the STM root folder GUI? 
Maybe if it works well, subfolders could fill entire screen width and maybe a horizontal scrollbar could be added for (rare) overlength files to match the vertical scroll bar?

**Comment by Hofknecht** [comment]
@chip33 
Thank you for starting this discussion!

1
yes normally that also works like this. There are 3 maximum values e.g. for height. 1. screen 2. height needed by menu 3 maximum value from options.
we changed maximum value from options from pixel to percentag. 100% now currently 450 of 1080 pixel (at 100% dpi) because it windows start menu is similar to that. When changing e.g. to 300% then screen as maximum is used. 
We could change that 100% is 1080 pixel (at 100% dpi). Then the default value should be something like 40% for height.
If you think we should adjust you can open a issue : ), is easy to implement

2
The second point with scrollbar for menu width is a big more tricky to implement, but sounds like a great feature. i make a note to analyze this more (https://github.com/Hofknecht/SystemTrayMenu/projects/17)

**Comment by chip33** [comment]
@Hofknecht 
thank you for the explanation. 
1. I had forgotten about the effect of DPI (Display Scaling) and the effect that has on changing the size. :) The way it is now is very nice when DPI is considered.
